Mage.Sets/src/mage/cards/c/ChissGoriaForgeTyrant.java

```diff
@@ -0,0 +1,213 @@
+        super(card);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrant copy() {
+        return new ChissGoriaForgeTyrant(this);
+    }
+}
+
+class ChissGoriaForgeTyrantEffect extends OneShotEffect {
+
+    ChissGoriaForgeTyrantEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it gains affinity for artifacts.";
+    }
+
+    private ChissGoriaForgeTyrantEffect(final ChissGoriaForgeTyrantEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantEffect copy() {
+        return new ChissGoriaForgeTyrantEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        // Return if no cards exiled, effect still applied
+        Cards cards = new CardsImpl(player.getLibrary().getTopCards(game, 5));
+        if (cards.isEmpty()) {
+            return true;
+        }
+        player.moveCardsToExile(cards.getCards(game), source, game, false, CardUtil.getExileZoneId(game, source), CardUtil.getSourceName(game, source));
+        cards.retainZone(Zone.EXILED, game);
+
+        Cards castableCards = new CardsImpl(cards.getCards(StaticFilters.FILTER_CARD_NON_LAND, game));
+        Set<MageObjectReference> morSet = new HashSet<>();
+        castableCards.stream()
+                .map(uuid -> new MageObjectReference(uuid, game))
+                .forEach(morSet::add);
+
+        game.addEffect(new ChissGoriaForgeTyrantCanPlayEffect(morSet), source);
+        game.addEffect(new ChissGoriaForgeTyrantAffinityEffect(morSet), source);
+        return true;
+    }
+}
+
+class ChissGoriaForgeTyrantCanPlayEffect extends AsThoughEffectImpl {
+
+    private final Set<MageObjectReference> morSet = new HashSet<>();
+
+    ChissGoriaForgeTyrantCanPlayEffect(Set<MageObjectReference> morSet) {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.EndOfTurn, Outcome.Benefit);
+        this.morSet.addAll(morSet);
+    }
+
+    private ChissGoriaForgeTyrantCanPlayEffect(final ChissGoriaForgeTyrantCanPlayEffect effect) {
+        super(effect);
+        this.morSet.addAll(effect.morSet);
+    }
+
+    @Override
+    public ChissGoriaForgeTyrantCanPlayEffect copy() {
+        return new ChissGoriaForgeTyrantCanPlayEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        UUID objectIdToCast = CardUtil.getMainCardId(game, sourceId);
+        return game.getCard(objectIdToCast).isArtifact(game) 
```

Comment:
Just tried that code. On starting cast of an exiled artifact, got that null exception.
![image](https://github.com/magefree/mage/assets/34709007/388b476e-0386-4a49-bea9-21a9c4ed3bd4)

Null-checking the card seemed to work. Did not test much more beyond that.

Reply:
Yep, any `game.getCard()` or `game.getPermanent()` should be null checked before calling methods on it